File: src/strategies/vwap_bounce.rs
use crate::{calculate_vwap, is_three_red_candles, Candle};

pub struct VwapBounce;

impl super::Strategy for VwapBounce {
    fn name(&self) -> &str {
        "VWAP Bounce"
    }

    fn check_signal(&self, candles: &[Candle]) -> bool {
        if candles.len() < 3 {
            return false;
        }
        let vwap = calculate_vwap(&candles);
        let last_close = candles.last().unwrap().close;
        last_close < vwap && is_three_red_candles(&candles)
    }
}

File: vwap_option_bot/strategies/vwap_bounce.py
from ..models import Candle
from ..utils import calculate_vwap, is_three_red_candles
from . import Strategy

class VwapBounce(Strategy):
    def name(self) -> str:
        return "VWAP Bounce"

    def check_signal(self, candles: list[Candle]) -> bool:
        if len(candles) < 3:
            return False
        vwap = calculate_vwap(candles)
        last_close = candles[-1].close
        return last_close < vwap and is_three_red_candles(candles)
